### ISSUE TITLE
Working with DVW

### ISSUE THREAD

**Issue by djones4487169** [ivmartel/dwv#362]
I'm considering using the DVW viewer as the basis for a new project I'm developing that looks at mammograms in particular image processing of digital images e.g., segmentation, contour analysis etc. Is the DVW source code completely customisable with regards, logos, colours, look-and-feel etc.?

I have many routines written in C++, OpenCV, Python that I would like to integrate into the DVW viewer. Is this possible? I know it is a JS, CSS, HTML5 library but are there workarounds?

Also, would it be possible to integrate some WebGL 3D style graphics into the DVW viewer? Do you know if anyone has done this?

Any help appreciated.

**Comment by ivmartel** [comment]
Yes, DWV is customisable, it provides functionality and gui that can be overwritten through js code and css. 

Intensive code should be executed on a server and then displayed in the viewer. Less intensive code can go with the viewer as you can see in the demos (basic thresholding, floodfill, filtering...). 

Adding some WebGL views should be possible. An example of a full WebGL viewer is the [ami](https://github.com/FNNDSC/ami) project.

**Comment by djones4487169** [comment]
Is it possible to have a customised version of DWV standalone web served
app that connects to Orthanc which sits in the background invisible until
requested? With Orthanc the DVW is a plug-in - is it possible to use
Orthanc externally to fetch data as and when from the DVW client interface?
If so, how would you do it?

Thanks
David

**Comment by ivmartel** [comment]
Yes, you can fork the dwv-orthanc-plugin to customise it and make Orthanc use it. I'm not sure I get the invisible part, Orthanc provides a web interface to the database and can use the plugin to view the data.

DWV is agnostic about the data storage, it displays what it is told to display. The aim is to integrate it in data providers such as a PACS the same way it is done in Orthanc (or others, see [Pacs-support](https://github.com/ivmartel/dwv/wiki/Pacs-support)).

**Comment by ivmartel** [comment]
I hope I answered your questions. Don't hesitate if you have others. I'm also interested in your feedback if you think DWV is missing some functionality.

**Comment by djones4487169** [comment]
Yes thanks!

I have been looking at several examples that make use of Cornerstone, orthanc etc. e.g., DVW, OHIF, etc. A quick question:

I can get the all the code from your repository but what IDE have you used that I can set up to make changes and easily see results without having to do complete builds? I'm thinking Eclipse but tried to get a project setup and did not work very well. I guess I need several extra plug-ins etc. Basically, how would you set up a development environment to work with meteor, node.js, cornerstone, javascript, css, html type projects such as yours?

David

**Comment by djones4487169** [comment]
s thanks!

I have been looking at several examples that make use of Cornerstone,
orthanc etc. e.g., DVW, OHIF, etc. A quick question:

I can get the all the code from your repository but what IDE have you used
that I can set up to make changes and easily see results without having to
do complete builds? I'm thinking Eclipse but tried to get a project setup
and did not work very well. I guess I need several extra plug-ins etc.
Basically, how would you set up a development environment to work with
meteor, node.js, cornerstone, javascript, css, html type projects such as
yours?

David

**Comment by ivmartel** [comment]
_(removed duplicate comment)_

**Comment by ivmartel** [comment]
For some notes on dev tools, check out this wiki page: [Developing](https://github.com/ivmartel/dwv/wiki/Developing). I used to use Eclipse but it got a bit big so I switched to using the Atom text editor with some linting plugin. I'm talking about DWV, I can't tell for Meteor and Cornerstone.

**Comment by djones4487169** [comment]
Thanks - I will give it try.

**Comment by ivmartel** [comment]
Closing for now. Feel free to reopen if needed.
